x-pack/plugins/enterprise_search/common/ml_inference_pipeline/index.ts

```diff
@@ -25,10 +26,30 @@ import {
   InferencePipelineInferenceConfig,
 } from '../types/pipelines';
 
-export const TEXT_EXPANSION_TYPE = SUPPORTED_PYTORCH_TASKS.TEXT_EXPANSION;
-export const TEXT_EXPANSION_FRIENDLY_TYPE = 'ELSER';
-export const ML_INFERENCE_PREFIX = 'ml.inference.';
-export const ELSER_MODEL_ID = '.elser_model_1';
+const ML_INFERENCE_PREFIX = 'ml.inference.';
+
+class InferenceModel {
+  public taskType: string;
+  public friendlyTaskType: string;
+  public modelId: string;
+
+  constructor() {
+    this.taskType = SUPPORTED_PYTORCH_TASKS.TEXT_EXPANSION;
+    this.friendlyTaskType = 'ELSER';
+    this.modelId = '';
+  }
+
+  public setModelId(modelId: string): void {
+    this.modelId = modelId;
+  }
+}
+
+export const elserInferenceModel = new InferenceModel();
+
+export const configureElserInferenceModel = async (elasticModels: ElasticModels) => {
+  const elserModel = await elasticModels.getELSER({version :2});
```

Comment:
ℹ️ Using the `getELSER` method allows to return the optimized version of model for your architecture.

Reply:
re-using the `ElasticModels` from the ml plugin is a good idea IMO :+1: